indico_jacow/controllers.py

```diff
@@ -270,3 +277,101 @@ def _process(self, file):
             'identifiers': identifiers,
             'unknown_emails': list(unknown_emails)
         })
+
+
+class RHBrevoAPI(RH):
+    def __init__(self):
+        from indico_jacow.plugin import JACOWPlugin
+
+        super().__init__()
+        configuration_brevo = brevo_python.Configuration()
+        configuration_brevo.api_key['api-key'] = JACOWPlugin.settings.get('brevo_api_key')
+        self.api_instance = brevo_python.ContactsApi(brevo_python.ApiClient(configuration_brevo))
+
+
+class RHMailingLists(RHUserBase, RHBrevoAPI):
+    def _process(self):        
+        valid_contact_ids = set()
+        emails = self.user.all_emails
+
+        for email in emails:
+            try:
+                contact_info = self.api_instance.get_contact_info(email)
+                contact_data = contact_info.to_dict()
+                if 'list_ids' in contact_data:
+                    valid_contact_ids.update(contact_data['list_ids'])
+            except ApiException as e:
+                print(f'Warning: Contact {email} not found in Brevo. Skipping...')
+                return {'error': e.reason}, e.status
+
+        # Get all Brevo lists
+        try:
+            api_response_2 = self.api_instance.get_lists()
+            response_2_dict = api_response_2.to_dict()
+        except ApiException as e:
+            print(f'Exception when calling ContactsApi->get_lists: {e.reason}')
+            return {'error': e.reason}, e.status
+
+        # Mark subscribed lists
+        for lst in response_2_dict.get('lists', []):
+            if lst['id'] in valid_contact_ids:
+                lst['subscribed'] = True
+            else:
+                lst['subscribed'] = False
+
+        # Convert to JSON and return
+        mailing_lists = json.dumps(response_2_dict)
+        return WPUserMailingLists.render_template('mailing_lists.html', 'mailing_lists', user=self.user,
+                                                  mailing_lists=mailing_lists)
+
+
+class RHMailingListSubscribe(RHUserBase, RHBrevoAPI):
+    def _process(self):
+        email = self.user.email
+        print(email)
+        try:
+            list_id = int(request.args.get('list_id'))
+        except (ValueError, TypeError):
+            return {'error': 'Invalid or missing list_id. It must be an integer.'}, 400
+
+        try:
+            self.api_instance.get_contact_info(email)
+            try:
+                contact_email = brevo_python.AddContactToList(emails=[email])
+                response = self.api_instance.add_contact_to_list(list_id, contact_email)
+                return response.to_dict(), 200
+            except ApiException as e:
+                print(f'Could not add contact {email} to list {list_id} due to {e}')
+                return {'error': 'Failed to subscribe to the list.'}, e.status
+        except ApiException as e:
```

Comment:
I don't like the idea of these nested handlers for the same exception. I suggest putting only the `self.api_instance.get_contact_info(email)` in the initial `try` block and then put the next operation after you're done with its `except` block.